README.md
# Kibana 6.8.1

## Configuration
Configuration is done via environment variables.

* `ELASTICSEARCH_URL`: ElasticSearch url. Default: `http://localhost:9200`.
* `ELASTICSEARCH_CERTIFICATE_VERIFICATION`: Whether to verify elasticsearch certificate. Default: `true`
* `ELASTICSEARCH_SSL_CA_CERT_PATH`: (Optional) Path to the PEM file for the certificate authority for your Elasticsearch instance.
* `ELASTICSEARCH_USERNAME`: (Optional) Elasticsearch username for Kibana. Default: `""`.
* `ELASTICSEARCH_PASSWORD`: (Optional) Elasticsearch password for Kibana user. Default: `""`.
* `ENABLE_KIBANA_RUNAS_HEADER`: (Optional) Allow Kibana to pass “run as” header to Elasticsearch from Nginx. Default: `false`.
* `KIBANA_SERVER_HOST`: Listen host. Default: `0.0.0.0`.
* `KIBANA_SERVER_PORT`: Listen port. Default: `5601`.
* `XPACK_SECURITY_ENABLE`: Whether to enable X-Pack security plugin. Default: `false`.
* `XPACK_SECURITY_ENCRYPTION_KEY`: Required if X-Pack security enabled. An arbitrary string of 32 characters or more used to encrypt credentials in a cookie. It is crucial that this key is not exposed to Kibana users. Default: (randomly generated UUID).
* `XPACK_SECURITY_SECURE_COOKIES_ENABLE`: Whether to enable secure cookies. Default: `false`. Read more [here](https://www.elastic.co/guide/en/x-pack/5.1/kibana.html#security-ui-settings)
* `XPACK_MONITORING_ENABLE`: Whether to enable X-Pack monitoring features. Default: `false`.
* `XPACK_MONITORING_REPORT_STATS`: Whether or not to send cluster statistics to Elastic. Default: `false`.
* `XPACK_GRAPH_ENABLE`: Whether to enable X-Pack graph features. Default: `false`.
* `XPACK_ML_ENABLE`: Whether to enable X-Pack machine learning features. Default: `false`.
* `XPACK_REPORTING_ENABLE`: Whether to enable X-Pack reporting features. Default: `false`.
* `XPACK_REPORTING_ENCRYPTION_KEY`: X-Pack reporting encryption key. Default: (randomly generated UUID).
* `XPACK_REPORTING_CSV_MAX_SIZE_BYTES`: X-Pack CSV report max size. Default: 10485760. (10MB)
* `XPACK_APM_ENABLE`: Whether to enable X-Pack APM module. Default: `false`.
* `XPACK_APM_UI_ENABLE`: Whether to enable X-Pack APM UI. Default: `false`.
* `XPACK_INFRA_ENABLE`: Whether to enable X-Pack Infra & Logs UI. Default: `false`.
* `XPACK_CANVAS_ENABLE`: Whether to enable X-Pack Canvas UI. Default: `false`.
* `XPACK_GROKDEBUGGER_ENABLE`: Whether to enable Grok debugger in Dev Tools. Default: `false`.
* `XPACK_SEARCHPROFILER_ENABLED`: Whether to enable Search profiler in Dev Tools. Default: `false`.
* `XPACK_UPTIME_ENABLE`: Whether to enable X-Pack Uptime UI. Default: `false`.
* `XPACK_MAPS_ENABLE`: Whether to enable X-Pack Maps UI. Default: `false`.

## Plugins

### X-Pack
See documentation: https://www.elastic.co/guide/en/x-pack/current/index.html


CHANGELOG.md
# CHANGELOG

## 6.8.1-1

* Upgrades base image to quay.io/ukhomeofficedigital/centos-base:v0.5.13
* Upgrades Kibana to version 6.8.1: https://www.elastic.co/guide/en/kibana/6.8/release-notes.html

* `X-pack` installed by default
* The following environment variables are REMOVED:
    * `XPACK_MONITORING_REPORT_STATS`
* The following environment variables are ADDED (See README.md for details):
    * `XPACK_TELEMETRY_ENABLE`
    * `KIBANA_SERVER_HOST`
    * `KIBANA_SERVER_PORT`
    * `XPACK_APM_ENABLE`
    * `XPACK_APM_UI_ENABLE`
    * `XPACK_INFRA_ENABLE`
    * `XPACK_CANVAS_ENABLE`
    * `XPACK_GROKDEBUGGER_ENABLE`

## 5.6.16-1

* Upgrades Kibana to version 5.6.16: https://www.elastic.co/guide/en/kibana/5.6/release-notes.html

### Notes:

The key upgrade in this version is the addition of the Upgrade Assistant and its API, which can be used to stage a migration to v6.x


run.sh
#!/usr/bin/bash

export HOME=/kibana

: ${ELASTICSEARCH_URL:=http://localhost:9200}
: ${ELASTICSEARCH_CERTIFICATE_VERIFICATION:=full}
: ${ELASTICSEARCH_SSL_CA_CERT_PATH:=}
: ${ELASTICSEARCH_USERNAME:=}
: ${ELASTICSEARCH_PASSWORD:=}
: ${ENABLE_KIBANA_RUNAS_HEADER:=false}
: ${XPACK_SECURITY_ENABLE:=false}
: ${XPACK_SECURITY_ENCRYPTION_KEY:=$(uuidgen)}
: ${XPACK_SECURITY_SECURE_COOKIES_ENABLE:=false}
: ${XPACK_MONITORING_ENABLE:=false}
: ${XPACK_TELEMETRY_ENABLE:=false}
: ${XPACK_GRAPH_ENABLE:=false}
: ${XPACK_ML_ENABLE:=false}
: ${XPACK_REPORTING_ENABLE:=false}
: ${XPACK_REPORTING_ENCRYPTION_KEY:=$(uuidgen)}
: ${XPACK_REPORTING_CSV_MAX_SIZE_BYTES:=10485760}
: ${XPACK_APM_ENABLE:=false}
: ${XPACK_APM_UI_ENABLE:=false}
: ${XPACK_INFRA_ENABLE:=false}
: ${XPACK_CANVAS_ENABLE:=false}
: ${XPACK_GROKDEBUGGER_ENABLE:=false}
: ${XPACK_SEARCHPROFILER_ENABLED:=false}
: ${XPACK_UPTIME_ENABLE:=false}
: ${XPACK_MAPS_ENABLE:=false}
: ${KIBANA_SERVER_HOST:=0.0.0.0}
: ${KIBANA_SERVER_PORT:=5601}

sed -e "s/%ELASTICSEARCH_CERTIFICATE_VERIFICATION%/${ELASTICSEARCH_CERTIFICATE_VERIFICATION}/" \
    -e "s/%XPACK_SECURITY_ENABLE%/${XPACK_SECURITY_ENABLE}/" \
    -e "s#%XPACK_SECURITY_ENCRYPTION_KEY%#${XPACK_SECURITY_ENCRYPTION_KEY}#" \
    -e "s/%XPACK_SECURITY_SECURE_COOKIES_ENABLE%/${XPACK_SECURITY_SECURE_COOKIES_ENABLE}/" \
    -e "s/%XPACK_MONITORING_ENABLE%/${XPACK_MONITORING_ENABLE}/" \
    -e "s/%XPACK_TELEMETRY_ENABLE%/${XPACK_TELEMETRY_ENABLE}/" \
    -e "s/%XPACK_GRAPH_ENABLE%/${XPACK_GRAPH_ENABLE}/" \
    -e "s/%XPACK_ML_ENABLE%/${XPACK_ML_ENABLE}/" \
    -e "s/%XPACK_REPORTING_ENABLE%/${XPACK_REPORTING_ENABLE}/" \
    -e "s/%XPACK_REPORTING_ENCRYPTION_KEY%/${XPACK_REPORTING_ENCRYPTION_KEY}/" \
    -e "s/%XPACK_REPORTING_CSV_MAX_SIZE_BYTES%/${XPACK_REPORTING_CSV_MAX_SIZE_BYTES}/" \
    -e "s/%XPACK_APM_ENABLE%/${XPACK_APM_ENABLE}/" \
    -e "s/%XPACK_APM_UI_ENABLE%/${XPACK_APM_UI_ENABLE}/" \
    -e "s/%XPACK_INFRA_ENABLE%/${XPACK_INFRA_ENABLE}/" \
    -e "s/%XPACK_CANVAS_ENABLE%/${XPACK_CANVAS_ENABLE}/" \
    -e "s/%XPACK_GROKDEBUGGER_ENABLE%/${XPACK_GROKDEBUGGER_ENABLE}/" \
    -e "s/%XPACK_SEARCHPROFILER_ENABLED%/${XPACK_SEARCHPROFILER_ENABLED}/" \
    -e "s/%XPACK_UPTIME_ENABLE%/${XPACK_UPTIME_ENABLE}/" \
    -e "s/%XPACK_MAPS_ENABLE%/${XPACK_MAPS_ENABLE}/" \
    -e "s/%KIBANA_SERVER_HOST%/${KIBANA_SERVER_HOST}/" \
    -e "s/%KIBANA_SERVER_PORT%/${KIBANA_SERVER_PORT}/" \
    -i /kibana/config/kibana.yml

if [[ -n "${ELASTICSEARCH_SSL_CA_CERT_PATH}" ]]; then
  sed -e "s#%ELASTICSEARCH_SSL_CA_CERT_PATH_CONFIG_ENTRY%#elasticsearch.ssl.certificateAuthorities: ${ELASTICSEARCH_SSL_CA_CERT_PATH}#" \
      -i /kibana/config/kibana.yml
else
  sed -e "s/%ELASTICSEARCH_SSL_CA_CERT_PATH_CONFIG_ENTRY%//" \
      -i /kibana/config/kibana.yml
fi

if [[ ${ENABLE_KIBANA_RUNAS_HEADER} == true ]]; then
  sed -e "s#%ELASTICSEARCH_REQUEST_HEADERS_WHITELIST_ENTRY%#elasticsearch.requestHeadersWhitelist: [ es-security-runas-user, authorization ]#" \
      -e "s#%XPACK_MONITORING_ELASTICSEARCH_REQUEST_HEADERS_WHITELIST_ENTRY%#xpack.monitoring.elasticsearch.requestHeadersWhitelist: [ es-security-runas-user, authorization ]#" \
      -i /kibana/config/kibana.yml

else
  sed -e "s/%ELASTICSEARCH_REQUEST_HEADERS_WHITELIST_ENTRY%//" \
      -e "s/%XPACK_MONITORING_ELASTICSEARCH_REQUEST_HEADERS_WHITELIST_ENTRY%//" \
      -i /kibana/config/kibana.yml
fi

if [[ -n "${ELASTICSEARCH_USERNAME}" && -n "${ELASTICSEARCH_PASSWORD}" ]]; then
  sed -e "s#%ELASTICSEARCH_USERNAME_CONFIG_ENTRY%#elasticsearch.username: ${ELASTICSEARCH_USERNAME}#" \
      -e "s#%ELASTICSEARCH_PASSWORD_CONFIG_ENTRY%#elasticsearch.password: ${ELASTICSEARCH_PASSWORD}#" \
      -i /kibana/config/kibana.yml
else
  sed -e "s/%ELASTICSEARCH_USERNAME_CONFIG_ENTRY%//" \
      -e "s/%ELASTICSEARCH_PASSWORD_CONFIG_ENTRY%//" \
      -i /kibana/config/kibana.yml
fi

exec /kibana/node/bin/node src/cli -e "${ELASTICSEARCH_URL}"
